### ISSUE TITLE
Implement new T&Cs for iRecord

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/iRecord#638]
Implement Indicia module - indicia licences

Photo licence option on user profile

Publish new T&C version at https://www.brc.ac.uk/irecord/terms_of_use

Test on dev iRecord

**Comment by burkmarr** [comment]
@johnvanbreda @BirenRathod The Indicia licenses module is on in irecord_dev and  Iform settings > Indicia licence choices is set to 'User can select licence for media (photos etc) only', but I can see no sign of a license selection control on a user profile page. Have I missed a configuration step or something?

**Comment by BirenRathod** [comment]
@johnvanbreda 
It hasn't appeared even in hidden fields list of user's manage field settings. Seems like there is still required some improvement of code or may be bug some where.

**Comment by johnvanbreda** [comment]
Not a bug, but I suspect you need to add some licenses to the iRecord website at http://devwarehouse.indicia.org.uk/index.php/website/edit/23 (Licences tab).

**Comment by burkmarr** [comment]
@johnvanbreda - spot on thanks.

**Comment by burkmarr** [comment]
@johnvanbreda - where is a user's license selection stored? I thought it would be in the Drupal DB alongside things like location, taxon_groups etc, but I don't see it there.

**Comment by burkmarr** [comment]
@johnvanbreda - I figured out that this is stored in the Warehouse users_websites table. I'm thinking that I can access the value for a user from PHP in the top pane of iRecord's homepage. Can you give me any pointers on the best way to access that value - do I need to use the client_helpers API?

**Comment by burkmarr** [comment]
@DavidRoy, @kitenetter and @johnvanbreda I have a note from our last meeting that once the media licensing option is implemented, I should check the response when masquerading as a user with many existing images who sets the media license option on their profile. I did this on irecord_dev and the response was immediate. A note at the bottom of the license options says "Select the licence to apply to your photos and other media. If you change the selected licence then _it will only affect media you enter from now on_" (my emphasis). That would explain the quick response, but it means that existing images are not updated with the selected license. What are your views on this?

**Comment by johnvanbreda** [comment]
@burkmarr The code in the iform_licences.module file is fairly clear about the intentions. The message shown if you have never set a licence before should tell you that it will affect all your existing records. The message you saw was the one that should be shown if you have already set a licence.

Maybe there was a bug then, in that it thinks you had already set a licence even though you hadn't? I'd be curious to know if the media records associated with this user had their licence updated?

**Comment by burkmarr** [comment]
@johnvanbreda - my bad. I masqueraded as the wrong user - one with no previous images - and I had only seen the instruction on revisiting the page. I've just done it for a user with over 13,000 images. The initial instruction was as you said and the page returned less than 1 minute after submission. I've checked the DB and the license_id was updated for all their images.

**Comment by burkmarr** [comment]
@johnvanbreda - just a prompt to ask if you have any pointers about the 'right' way to go about implementing a message on home page to users who have not yet set a media licensing option.

**Comment by johnvanbreda** [comment]
@burkmarr - I guess the right way would be for the licensing module to include a login hook which adds a nagg message after you log in if there is no setting for the licence. I don't think I have any existing code to do this.

**Comment by johnvanbreda** [comment]
> I figured out that this is stored in the Warehouse users_websites table. I'm thinking that I can access the value for a user from PHP in the top pane of iRecord's homepage. Can you give me any pointers on the best way to access that value - do I need to use the client_helpers API?

You can hopefully use data_entry_helper::get_population_data to pull data out - there are plenty of examples if you search the code. This uses the views in the database (normally list_* or detail_* depending on the level of detail required) to access data in the tables. Views are used to join in info from other tables so the information is useful without multiple hits on the database.

In this case, I note that both `list_users` and `detail_users` contain the licence_id column but presumably have not been updated to include the media_licence_id column. I would advocate for updating these views using an upgrade script. Then you can pull data from list users by setting the table parameter for get_population_data to 'user'.

**Comment by burkmarr** [comment]
Merge request: https://github.com/Indicia-Team/drupal-7-module-iform/pull/15

**Comment by burkmarr** [comment]
@johnvanbreda @BirenRathod - when this this merged into the dev branch I'll can update the and get @DavidRoy and @kitenetter to check it out.

**Comment by BirenRathod** [comment]
@burkmarr 
I leave that to @johnvanbreda to release code on iRecord.

**Comment by johnvanbreda** [comment]
Code is released now.

**Comment by burkmarr** [comment]
@DavidRoy @kitenetter. This change to the iForm module is implemented on iRecord Dev now. If you masquerade as any user without media license options set, you should see a message at the top of the screen to instruct to set media licensing. If you are happy with this, we can implement on master so that it goes into live site when next updated. Then we can update the T&Cs page too.

**Comment by kitenetter** [comment]
On the user account, under the licence choices, it says:
"Select the licence to apply to your photos and other media. THIS LICENCE WILL ALSO BE APPLIED TO ALL YOUR PREVIOUSLY ENTERED RECORDS ON IRECORD DEVELOPMENT SITE."

I would prefer to make this explicitly limited to photos not records, and suggest:
"Select the licence to apply to any photos and other media that you upload. THIS LICENCE WILL ALSO BE APPLIED TO ALL THE PHOTOS/MEDIA YOU HAVE PREVIOUSLY UPLOADED TO IRECORD [DEVELOPMENT SITE]."

I think we probably need one more round of checking the wording on the new T&Cs page, can you remind me where the text is (unless @DavidRoy would prefer to stick with what we've got).

**Comment by burkmarr** [comment]
Yes - I think that message is just plain wrong. I think the license is applied to previous images, but not to records.

The wording of the new T&Cs is on this unpublished page: https://www.brc.ac.uk/irecord/node/7801

My understanding is that when we are ready to go, we will update the current T&Cs page to this new wording and then delete this unpublished page.

**Comment by burkmarr** [comment]
I also noticed a typo in the banner message: 

> [...] This limits to usefulness of your images. [...]

'to' should be 'the'.

**Comment by burkmarr** [comment]
@kitenetter - I've pulled the updated develop branch into http://test.brc.ac.uk/irecord_dev so the wording change you requested on the user account page is now effective there.

**Comment by burkmarr** [comment]
@kitenetter - typo in info message that I mentioned a couple of days ago is now fixed in develop branch (and on http://test.brc.ac.uk/irecord_dev/).

**Comment by johnvanbreda** [comment]
@burkmarr code now all live. Does any configuration need to be done before this can be closed?

**Comment by DavidRoy** [comment]
I've just looked at this on irecord_dev.  A few comments.
1. should the link to your profile page, link direct to the editing version of the account profile page (rather than the view) and ideally direct to the licence choices. e.g. link in this text
**You have not yet specified a media licence. This limits the usefulness of your images. To set a licence, edit your profile page, and select an option.**
2. should the list of licences link to their description, e.g. for this list (@kitenetter are we offering all these options?)
Licence for your photos and other media *
 CC0: Creative Commons "No Rights Reserved"
 OGL: Open Government licence (UK)
 CC BY: Creative Commons Attribution
 CC BY-NC: Creative Commons Attribution-NonCommercial
 CC BY-SA: Creative Commons Attribution-ShareAlike
 CC BY-ND: Creative Commons Attribution-NoDerivatives
 CC BY-NC-SA: Creative Commons Attribution-NonCommercial-ShareAlike
 CC BY-NC-ND: Creative Commons Attribution-NonCommercial-NoDerivs
3. reword help text to:
Select the licence to apply to your photos and other media. We strongly recommend selecting a CC0 licence option as this helps support the use of your photos in research and conservation.  We do not recommend that commercially valuable photos are uploaded to iRecord.  THIS LICENCE WILL ALSO BE APPLIED TO ALL THE PHOTOS AND OTHER MEDIA YOU HAVE PREVIOUSLY UPLOADED TO IRECORD DEVELOPMENT SITE.
4. should we create a news item/blog to link to in order to explain the value of photos and what we recommend (i.e. linked to the point 3 above)
5. I've selected a licence option, but I can't test whether it then applied to my previous photos as they aren't viewable on iRecord-dev

**Comment by burkmarr** [comment]
@johnvanbreda yes it does, but I'll hold off in order to address @DavidRoy's concerns.

Config steps (a reminder to myself):

- From the Warehouse, add required license options to website
- From Drupal Indicia module config, turn on Indicia Licenses
- From Iform settings > set Indicia licence choices to 'User can select licence for media (photos etc) only'

**Comment by burkmarr** [comment]
@DavidRoy:

Changing the link to the user's profile edit shouldn't be a problem and neither should changing the wording of the help text. 

You can run a query on the dev warehouse occurrence_media table and examine the value of the licence_id column to see if your images have been updated. (I earlier confirmed that this was the case for a user with some 13,000 images.)

I think that links to the license definitions should be doable - the license defintion URLs are already stored in the license table. If fact there are two for each license - one links to the legal definition and one to the 'readable' defintion (e.g. for CC BY, legal is https://creativecommons.org/licenses/by/4.0/legalcode and readable is https://creativecommons.org/licenses/by/4.0). Would you prefer to link to the readable definition?

**Comment by DavidRoy** [comment]
Thanks @burkmarr 

Linking to the readable version is best I think.

For deploying to live is it possible to do this in stages.
1. implement licence choice options in user account settings.  [I can then test that it updates my images]
2. implement message for users to choose a licence

**Comment by burkmarr** [comment]
@DavidRoy - it is possible to do it in two stages, but not through configuration as things stand; it would require a two-stage code update.

Any other users who update their profile after stage 1 of the update would also be able to select a media licence option. In fact this is a mandatory field, so no-one would be able to save changes to their profile without selecting a media licensing option. 

I wonder if it wouldn't be better to test it on dev instead? The live 'uploads' folder that contains all the photos is huge so I guess just copying that onto dev is not an option? But you could masqeurade as any random user on irecord_dev, add some pictures (at this stage without a licence) then go and set their media licence option and see if it all looks as you'd like it.

**Comment by burkmarr** [comment]
(You won't be able to do this using the user you logged in as last time because, as it's a madatory field, you cannot unset the licence option.)

**Comment by DavidRoy** [comment]
@burkmarr 
Agreed.  I did as you suggested and all worked as expected.  A few comments:
- in the explore-my records, photos tab the licence does not appear if the photo is clicked on
- from explore page, click on 'edit' record, the licence does not appear if photo clicked on
Ideally, the licence gets picked up on any use of photos

I'm happy for this to go live once @kitenetter confirms the communications plan for releasing this

**Comment by burkmarr** [comment]
@DavidRoy I've done the coding necessary to make the changes described in the first three of your five bullets above. These are in a pull request targeted on dev (https://github.com/Indicia-Team/drupal-7-module-iform/pull/18). I made a slight change to the text in your bullet 3 to remove the reference to iRecord so it just says 

> [...] We do not recommend that commercially valuable photos are uploaded. [...]

Addressing your comments about where the licences are displayed will require changes to  areas of the code outside the licensing module so I suggest that we raise separate issues for these. Perhaps we can cover this in Friday's meeting.

**Comment by burkmarr** [comment]
@DavidRoy the changes have been merged into the develop branch and are now reflected in irecord_dev if you want to see them.

**Comment by DavidRoy** [comment]
@burkmarr I tested again by masquerading as Bjorn.  I hit an issue when clicking on the link in the message about setting a licence (shown on home page).

The link was: http://test.brc.ac.uk/irecord_dev/user/user/6/edit#licenceopts

screenshot of error page:
![err](https://user-images.githubusercontent.com/2699478/66211279-98e21a80-e6b3-11e9-8374-8fe74f2e0142.png)

**Comment by burkmarr** [comment]
@DavidRoy  the 'user/user' part is the clue here - it should just be user. I think its because I've specified a relative path so whether or not it works depends on where you did the masqerade. I'll take a look. (The pink error notice is nothing to do with it - that's something I've noticed on irecord_dev for a few days.)

**Comment by burkmarr** [comment]
I've created a pull request to fix this @DavidRoy. I think that this was something to do with masquerading and should be alright under normal circumstances (if you navigate to homepage before masquerading it will work), but could potentially be a problem in other circumstances so best to fix it.

**Comment by burkmarr** [comment]
https://github.com/Indicia-Team/drupal-7-module-iform/pull/19

**Comment by burkmarr** [comment]
Are you happy to merge the above pull request @johnvanbreda?

**Comment by johnvanbreda** [comment]
@burkmar this is now deployed, I'll leave you to check and close.

**Comment by burkmarr** [comment]
@DavidRoy and @kitenetter - all the code changes required to implement media licensing have now been deployed to the live iRecord site, but it requires configuring to take effect, specifically:

- From the Warehouse, add required license options to iRecord website
- From the Drupal Indicia module config on iRecord, turn on Indicia Licenses
- From iRecord Iform settings > set Indicia licence choices to 'User can select licence for media (photos etc) only'

That's a five minute job.

@DavidRoy previously wrote:
> I'm happy for this to go live once @kitenetter confirms the communications plan for releasing this

There was also [a question](#issuecomment-536642271) about which of the available licensing options on the Warehouse to add to iRecord (first of the steps above).

Finally, once this is done, we were also going to update the wording of the T&Cs (https://www.brc.ac.uk/irecord/terms_of_use) to the new wording which we currently have stored on this unpublished page: https://www.brc.ac.uk/irecord/node/7801

**Comment by kitenetter** [comment]
@burkmarr can you remind me about the licence options, are these the options for media or for records or for both? (I think I'm correct that records will be treated as CC0 in all cases?)

For the T&Cs I need to review the text again following recent discussions with Quentin Tucker at CEH, will try to get on to that this week.

**Comment by burkmarr** [comment]
The licenses to be made available are set on a per-website basis (via Warehouse website config). 

![image](https://user-images.githubusercontent.com/9275178/67687971-04589880-f991-11e9-865a-0914e6de5a71.png)
Then in the iForm configuration on iRecord, once the licensing module is enabled, we will see the following options:

![image](https://user-images.githubusercontent.com/9275178/67687533-58af4880-f990-11e9-989d-3a269c18649d.png)
(This image is from irecord_dev)

On the configuration side there's no separate selection of licenses for records and media and my understanding is that if/when we enable the licensing for records, the licensing options available to the user for records will be the same as the licensing options available to them for media.

**Comment by kitenetter** [comment]
Okay. Since the iRecord terms and conditions explicitly state that records will be assigned a CC0 licence, then the licence options we are discussing will be for images only, and we will only ever implement the middle of the three radio button choices, as displayed above.

Are we proposing that managers of other website partitions within our data warehouse will be able to apply a different choice, i.e. to allow record-level licensing? If so I think a number of issues arise over how we share data with schemes and the NBN Atlas.

**Comment by kitenetter** [comment]
The current wording explaining the licences is:

> Select the licence to apply to your photos and other media. **We strongly recommend selecting a CC0 licence option as this helps support the use of your photos in research and conservation. We do not recommend that commercially valuable photos are uploaded.** THIS LICENCE WILL ALSO BE APPLIED TO ALL THE PHOTOS AND OTHER MEDIA YOU HAVE PREVIOUSLY UPLOADED TO IRECORD DEVELOPMENT SITE.

I don't think we should be "strongly recommending" that users give up all rights in their photos - that choice is up to the user, and personally I think a CC BY licence should not be seen as obstructing wider use in research and conservation. Also, if a user wishes to upload a commercially valuable photo under a suitable licence I don't see why we should seek to discourage them - again, it is their choice.

I'd prefer:

> Select the licence to apply to your photos and other media. **If you are happy to select a CC0 or CC BY licence option that will provide the best chance for your photos to be used in research and conservation.** THIS LICENCE WILL ALSO BE APPLIED TO ALL THE PHOTOS AND OTHER MEDIA YOU HAVE PREVIOUSLY UPLOADED TO iRECORD DEVELOPMENT SITE.

@DavidRoy what's your view on this?

**Comment by kitenetter** [comment]
The list of licences that we currently display on the dev site is:

![image](https://user-images.githubusercontent.com/16039111/67696680-32dd7000-f99f-11e9-96bd-2cba52e699fe.png)

This is close to the options available on Flickr, except for:
- we include OGL
- we don't have an option for no licence (i.e. all rights reserved)
- we don't have "Public Domain Work" - my understanding is that this is an option whereby it is declared that no licence restrictions can be found, and that therefore the image is regarded as public domain until proven otherwise (Flickr also has "Public Domain Dedication", which is equivalent to CC0, the distinction being that in the case of CC0 the public domain has been explicitly declared by the copyright owner, as opposed to being assumed in the absence of indications to the contrary)

Suggested actions:
1. Remove the OGL option from our list (on the grounds that it is not applicable to individual users)
2. Add an option for "None (all rights reserved)"

As things stand on the dev site we don't offer a default, pre-filled choice, and I think that is the correct approach.

**Comment by DavidRoy** [comment]
@kitenetter as discussed, I'm happy with your rewording of the licences and agree on aligning to Flickr.

For some surveys, we can auto-populate the image licences as the terms and conditions (i.e. iRecord Butterflies app T&Cs are open?)

I think we allow occurrence-level licences to vary for other Indicia websites.  If needbe, we might have to divide up datasets by licence when sharing with NBN.

**Comment by kitenetter** [comment]
@DavidRoy we do have occurrence-level licences for some websites, although arguably they have never been implemented fully and also arguably they should perhaps be discouraged. 

But the point is that if the system remains as described by Rich above, it only allows one set of licence options to be set up, which might then be offered (in some websites) for both media and records. That is likely to make trouble for all concerned, since the filtering needed will get complex (some licences are allowed by NBN, some not, and how do we handle conflicts between occurrence-level licences and dataset-level licences?). 

Also, what does it actually mean to say that an individual occurrence is "Attribution-NonCommercial-NoDerivs"?

Regarding the apps, they do have terms that are similar to the CC0 licence, but they don't explicitly say that a CC0 licence is being applied. I don't know if we are justified in applying a CC0 licence in this way. I will try to find out.

**Comment by DavidRoy** [comment]
By having the Indicia data model allowing occurrence-level licences we can't prevent this being used for websites out of our control?  But I agree with the general principle that it should avoided due to downstream complexity, and we can take a simple approach for websites directly in our control.  

This issue is only covering iRecord so we don't need to get overly bogged down in the wider issue?

I would welcome a review of the app T&Cs (when surveys under the iRecord website) regarding images.  It's likely they don't explicitly mention CC-licenses but I'm prepared to make a call on how we interpret them and accept the risk.

**Comment by burkmarr** [comment]
We can ensure that the OGL option is not available in the the iRecord - this is a configuration step on the warehouse for the iRecord website.

But the other changes will require further changes to the Licensing module:

1. Add the licensing option "None (all rights reserved)"
2. Change the licensing selection text - [...] If you are happy to select a CC0 or CC BY licence option that will provide the best chance for your photos to be used in research and conservation. [...]

The first one needn't affect another other websites since the new licensing option will only appear on a website if explicitly selected for it in the warehouse configuration. But the second one will affect all Indicia websites that have the Licensing module implemented for images. So I just want to check with everyone that this form of words is generally applicable within the broad Indicia strategy, i.e. that it is not specific to iRecord. My feeling is that it *is* generally applicable.

**Comment by DavidRoy** [comment]
Thanks.  Licence selection text _is_ generally applicable in my view

**Comment by burkmarr** [comment]
Pull request made to implement wording change: https://github.com/Indicia-Team/drupal-7-module-iform/pull/20

New 'All Rights Reserved' license option added to Warehouse via the Warehouse GUI: http://warehouse1.indicia.org.uk/index.php/licence/edit/9

**Comment by johnvanbreda** [comment]
@burkmarr the deployment of the code is now done.

**Comment by burkmarr** [comment]
Many thanks @johnvanbreda.

**Comment by burkmarr** [comment]
The media licensing is now ready to be configured. @kitenetter was going to review the new T&Cs before configuring. (See this previous comment: https://github.com/BiologicalRecordsCentre/iRecord/issues/638#issuecomment-546969557.)

**Comment by kitenetter** [comment]
Still struggling to find time to address this one.

**Comment by kitenetter** [comment]
@burkmarr I'm pretty sure that when I went to register a new account on the dev site last week it displayed some terms of use text that I had to agree to. I wanted to revisit that text, but I can no longer find it if I attempt to register another new account. Has it been lost from dev?

I have a copy of the text in Google docs (and will share shortly) but I wanted to double-check how it appears in context on this site and that's what I can't find.

**Comment by burkmarr** [comment]
@kitenetter. I don't think that users have ever been forced to check a box or anything to agree to terms and conditions. But I have a recollection that we agreed it wasn't necessary (at the team meeting in Wallingford when we began discussing this). We could perhaps do with a link to the T&Cs, e.g. on the registration page, but this hasn't been specified. (The links to the current T&C page and the original new draft text are in this comment: https://github.com/BiologicalRecordsCentre/iRecord/issues/638#issuecomment-536988369.)

**Comment by kitenetter** [comment]
@burkmarr as usual I think it may be my memory that is at fault here, apologies. We do ask people to tick a box to agree terms on the live website:
https://www.brc.ac.uk/irecord/user/register

And I think it was that page that I was trying to find, but for some reason I was convinced that I had found it on the dev site, not the live. But now I've actually looked at the live, it is there!

**Comment by burkmarr** [comment]
@kitenetter - I didn't know about that. I've looked at it and it's displayed because there is a 'terms of use' module which takes it's text from a page which you specify in its configuration. So it should be easy to update. That module is disabled on dev. I can enable it if you want to experiment. (Although I'm not sure about the value of getting people to explicitly tick the terms and conditions. And how do you handle changes to T&Cs for people that have already checked the box? I suspect the module won't do this - but I could be wrong.)

**Comment by johnvanbreda** [comment]
@burkmarr elsewhere I've used the Entity Legal module which does handle asking users to re-accept terms if they change.

**Comment by kitenetter** [comment]
Revised terms drafted, awaiting sign-off.

**Comment by kitenetter** [comment]
I've added the new text to two unpublished pages:
- Terms: https://www.brc.ac.uk/irecord/node/7801
- Privacy: https://www.brc.ac.uk/irecord/node/8097

One inconsistency is that in the Terms we say iRecord is run by BRC, which is within UKCEH, whereas on the Privacy Notice we say it is run by UKCEH, through the BRC work programme - which do you prefer @DavidRoy?

I think the next steps are:
1. Set up the registration form so that new users are given the link to the new pages and can tick a box to confirm acceptance of the terms (registration can only proceed if this box is ticked). Not sure if this tick box will appear on the inital registration page (i.e. before the email confirmation link has been sent) or should appear after clicking the email link.
2. Remove the current "under 16" tick box, and add a tick box for users to confirm they are aged 13 or over (registration can only proceed if this box is ticked).
3. On the dev site, reinstate the tools for users to allocate CC licences to images so we can have a final check for that before going live.
4. Set up procedure so that each iRecord user is asked to review the terms of use and the privacy notice next time they log in – my understanding is that we do not need existing users to opt-in via a tick box, as long as they are made aware of the new terms their continued use of the website will be taken as acceptance of those terms
5. Publish the new Terms and revised Privacy Notice.
6. Publish a Help page on the CC licences for images.
7. Announce in forums, and probably on the home page.

There may be other actions we need to take to alert other Indicia warehouse users to these changes.

**Comment by burkmarr** [comment]
@kitenetter - I've added a few licenses for the iRecord website on the Dev Warehouse. Go here to configure that: http://devwarehouse.indicia.org.uk/index.php/website/edit/23?tab=Licences. You can add/remove license options from there.

On the dev iRecord iForm configuration page, under 'Indicia license choices', I selected the option 'user can select licence for media (photos etc) only'.

I've removed the under 16 checkbox from the dev iRecord registration page and added two new mandatory checkbox controls to indicate that they are 13 or over and have read the terms and the privacy pages. Because they are mandatory, they can't submit the form if they don't check these boxes. I've included links to the Terms and Privacy pages in the help text for the terms checkbox (under the checkbox). That's the only way I know to configure the form to include the links. (These links on dev are pointing to the pages on the live site.)

I think that the registration form is the best place for these madatory checkboxes. It was easy to configure and it's the best way to prevent registration if the prospective user doesn't meet the prerequisites. It avoids disappointment later if they are not willing to accept the T&Cs.

Making these new checkboxes mandatory means that an existing user who wants to change any of their registration options cannot do so without checking the two boxes (and selecting a media license which is also mandatory).

**Comment by kitenetter** [comment]
Thanks @burkmarr. Just to try things out, I've updated the list of licences on the dev site to the list that we agreed earlier in this issue:

![image](https://user-images.githubusercontent.com/16039111/88531337-995cf080-cffa-11ea-8a48-6d0301704401.png)

On the "Prerequisites" section of the user account, we have this text:

> Confirm that you have read Terms and Privacy agreements
> You must confirm that you have read both the Terms and Conditions and the Privacy Agreement for using this site. Checking this box to ndicate that you agree to the terms and conditions and the Privacy agreement.

Users need to formally accept the terms, but I don't think they need to formally accept the Privacy Notice (which forms part of the terms). Also, we are using "Terms of use" (rather than "Terms and Conditions") and "Privacy Notice" (rather than "Privacy Agreement"). So I think the text should be:

> Confirm that you accept the Terms of use
> Check this box to confirm that you agree to the **Terms of use** for this website. Please also read our **Privacy Notice**.

The new user registration form on the Dev site includes the photo licences plus prerequisites and seems clear to me, although it would be good not to have separate tabs for the prerequisites and the "Who you are" information.

And this is stating the obvious, but the registration form on the live site is laid out slightly differently (and has a summarised set of terms plus an age limit both of which need to be removed and replaced by the new prerequisites). 

I will draft a Help page and info for the forums ASAP and then we should be ready to make this live.

The only remaining question that I can see in the above discussions relates to the images from the iRecord apps. As discussed at https://github.com/BiologicalRecordsCentre/iRecord/issues/638#issuecomment-547043471 some of the iRecord apps have a statement that photos submitted via the app can be used by the project and its partners. However, when people assign photo licences to their iRecord account that licence is applied to all their images, so there is potentially a conflict here. @DavidRoy I don't think we should delay implementing the new terms on the website, but this is a loose end that we will need to address.

**Comment by burkmarr** [comment]
@kitenetter - I wasn't aware of that on the live site. It has a terms and conditions module that wasn't enabled on the dev site. I'll enable it and take a look.

**Comment by burkmarr** [comment]
@kitenetter - I've enabled the terms and conditions module on dev. The configuration of that module points to a Drupal page from which it takes the text for the registration page. On dev that is currently http://test.brc.ac.uk/irecord_dev/terms_of_use. So editing that page will affect the text on the registration page.

The live site also currently gets is text from 'https://www.brc.ac.uk/irecord/terms_of_use'. When we want to change over to the new terms we can either point the T&C module to your new page or, if that is too much, update the current terms_of_use page that it points to, or create a new page with a summary of the new T&Cs and point to that.

**Comment by kitenetter** [comment]
@burkmarr I wanted to avoid having to **_summarise_** the main Terms of use page if we can, so that the Terms exist in one definitive place and we don't have to update it elsewhere. When the new terms go live they will be at https://www.brc.ac.uk/irecord/terms_of_use so from what you say the full version of the new terms will then appear on the registration page without us needing to take further action.

That is probably okay, although the new terms are longer than the old ones and more scrolling will be needed to reach the tick box and thus proceed to register.

Am I correct that the alternative would be to set up a completely new page (e.g. /terms_of_use/registration) and point the registration form to that? In which case the new page could simply give a link to the full terms without actually showing them, and in that case the registration page would look more similar to the view you originally had on dev?

But initially lets stick with the full set of terms on the verification page and see how it looks when we go live.

**Comment by burkmarr** [comment]
I'll replicate the new page on dev so we can see.

**Comment by burkmarr** [comment]
@kitenetter - I've done that and it looks good to me. When you say 'more scrolling will be needed to reach the tick box and thus proceed to register' - I presume you meant to say 'more reading...' ;). We can't force them to read the T&Cs but I think the least we should do is force them to scroll past!

**Comment by kitenetter** [comment]
Current situation from @burkmarr:

> It’s just configuration tasks now:
> 1. Configure iRecord on the Warehouse to add the required licenses - http://warehouse1.indicia.org.uk/index.php/website/edit/23
> 2. Once those licenses are added, the option to enable media licensing will appear in the iRecord configuration (just underneath the master species checklist selection as it is on dev): https://www.brc.ac.uk/irecord/admin/config/iform/settings
> 
> I think that’s it then. When a user logs in they will see a reminder about selecting a license (in green at the top of the page, just before ‘Log in successful’ message).  You can log into dev iRecord as yourself to see what that looks like. And on the profile page there will be the new message warning that if they have large numbers of images, it will take a little while.

Question remaining from me:
What do we need to do to update the new user registration page on the live site?
 
Once that is clarified I think I can proceed to carry out steps 1 and 2 above, and to publish the updated terms and privacy notice.

**Comment by kitenetter** [comment]
@burkmarr to check how age-limit tick box was set up on dev site registration page.

**Comment by burkmarr** [comment]
@kitenetter - sorry Martin I forgot all about the 'Over 13' checkbox step, so this field needs to be added to the main iRecord user account settings page. 

I can do that if you like but I need to just check somethign with @johnvanbreda first. John when I did this on dev I created a new field. At the meeting you suggested just changing the display of the current 'Under 16' checkbox. Couldn't there be a problem with that if someone has that checked - changing it's meaning could be problematic? Should I create a new field and hide the other one? The new checkbox is a madatory field so that new users will not be able to register unless they check the box to say that they are over 13.

@kitenetter - I also forgot that the Indicia licenses module must first be activated before the option to select the type of licensing for the website will appear on the Indicia settings page. I will also do that once I've sorted out the over 13 checkbox.

**Comment by burkmarr** [comment]
@kitenetter - the other thing we discussed at the meeting was where the T&C text is picked up from on the user account page. I'd forgotten how that works - you set it here: https://www.brc.ac.uk/irecord/admin/config/people/terms_of_use

So you can just switch this to point at your new page. I think you may need to use the node ID if the title of the new page is the same as the old page.

**Comment by johnvanbreda** [comment]
@burkmarr It sounds like the new field is different in functionality to the old one so I agree with what you've done on dev, i.e. create a new field and hide the old one.

**Comment by burkmarr** [comment]
Thanks @johnvanbreda. @kitenetter  - it's probably best if I implement these changes all at once:

- Create new new Over 13 field on the account settings page (https://www.brc.ac.uk/irecord/admin/config/people/accounts) and hide Under 16 checkbox.
- Ensure that the incorporated terms of use points to the new page (https://www.brc.ac.uk/irecord/admin/config/people/terms_of_use)
- Enable the licensing module (https://www.brc.ac.uk/irecord/admin/modules)
- Turn on licensing for media only (https://www.brc.ac.uk/irecord/admin/config/iform/settings)

If you can confirm that you are happy to go ahead and remind me where the new T&Cs are I will carry out these tasks on Wednesday morning.

**Comment by kitenetter** [comment]
Please do go ahead @burkmarr.

Current situation:
Alias :: node
/terms_of_use :: /node/20
/privacy-notice :: /node/7517

New situation:
Alias :: node
/terms_of_use :: /node/7801
/privacy-notice :: /node/8097

I think you only need the Terms of use page link for the registration page.

**Comment by burkmarr** [comment]
@kitenetter - this is all done but there is one minor problem. @BirenRathod - can you help with this? I cannot work out how to hide the old 'under 16' field from the account settings - I've relegated it to the 'hidden' category (see below) but it still appears. 

![image](https://user-images.githubusercontent.com/9275178/89995004-ceb34080-dc80-11ea-8a7c-12f3bd3940fd.png)

![image](https://user-images.githubusercontent.com/9275178/89994906-a9becd80-dc80-11ea-8d97-e9b00c6544a1.png)

**Comment by burkmarr** [comment]
@BirenRathod - I've worked it out. Fixed now.

**Comment by burkmarr** [comment]
@kitenetter - I also published the new privacy page and added redirects to this and the new terms of use page from the URLs for the old pages. So any links to the previous terms of use and privacy pages should now go to the new pages.

**Comment by kitenetter** [comment]
Many thanks @burkmarr - it's all looking good as far as I can tell. Will monitor it today but we should be able to close this one soon.

**Comment by kitenetter** [comment]
@burkmarr I've just gone to my own profile page to set a photo licence, and I'm still seeing both the 13 and 16 age questions.

**Comment by BirenRathod** [comment]
@kitenetter 
Would hat be ok if I delete it? It is not possible to hide on Drupal 7.

**Comment by burkmarr** [comment]
Ah - I removed it from the registration page but clearly not the profile view. I didn't realise that it was not possible to remove from the profile view without deleting.

**Comment by kitenetter** [comment]
@BirenRathod assuming @burkmarr is in agreement then yes I think we should remove the "Under 16" section from the profile form. (I assume that the existing data in the database will remain.)

**Comment by BirenRathod** [comment]
@kitenetter 
I have deleted that field on Drupal but will remain on warehouse database.

**Comment by kitenetter** [comment]
So far all seems well, good work everyone! Closing.
